client/js/components/procedure/StatementSegmentsList/StatementSegmentsEdit.vue

```diff
@@ -367,14 +377,29 @@ export default {
     },
 
     updateSegmentText (segmentId, val) {
-      const updated = { ...this.segments[segmentId], ...{ attributes: { ...this.segments[segmentId].attributes, ...{ text: val } } } }
+      let fullText = val
+      if (this.transformedText && this.transformedText !== fullText) {
+        fullText = this.transformedText
+      }
+      const updated = { ...this.segments[segmentId], ...{ attributes: { ...this.segments[segmentId].attributes, ...{ text: fullText } } } }
+
```

Comment:
This could be simplified (please make sure indentation is correct):

```suggestion
      const fullText = this.transformedText && this.transformedText !== fullText ? this.transformedText : val
      
      const updated = {
        ...this.segments[segmentId],
        attributes: {
          ...this.segments[segmentId].attributes,
          text: fullText
        }
      }
```